magefile.go

```diff
@@ -181,3 +196,147 @@ func licenser(mode licenserMode) error {
 
 	return nil
 }
+
+type Benchmark mg.Namespace
+
+func InstallBenchStat() error {
+	err := gotool.Install(gotool.Install.Package(goBenchstats))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (Benchmark) Run(ctx context.Context, outputFile string) error {
```

Comment:
I wonder if there's a way to make this a tad more user friendly, since if you omit the output file, you get a rather unhelpful `not enough arguments for target "Benchmark:Run", expected 1, got 0` error message from mage. Just adding a godoc that says something like `// Run all benchmarks. Takes the name of an output file as an argument` might help. Ditto with `benchmark:diff`

Reply:
please see [66c329c](https://github.com/elastic/elastic-agent-shipper-client/pull/22/commits/66c329c2fd2b0f34ecfd6252edab1906995ddb93) if makes this part any better.